classes/Product.php

```diff
@@ -6945,11 +7647,19 @@ public function modifierWsLinkRewrite()
         return true;
     }
 
+    /**
+     * @return array
+     */
     public function getWsProductBundle()
     {
         return Db::getInstance()->executeS('SELECT id_product_item as id, id_product_attribute_item as id_product_attribute, quantity FROM ' . _DB_PREFIX_ . 'pack WHERE id_product_pack = ' . (int) $this->id);
     }
```

Comment:
https://github.com/PrestaShop/PrestaShop/blob/41951d81761cbed89c711c3be96073c091ad1e92/classes/db/Db.php#L574-L632

Reply:
See the return type

Reply:
https://github.com/PrestaShop/PrestaShop/pull/13008#discussion_r269555970

Reply:
I do not agree with it.
The code should look like this:

```
    /**
     * @return array
     */
    public function getWsProductBundle()
    {
        $result = Db::getInstance()->executeS('SELECT id_product_item as id, id_product_attribute_item as id_product_attribute, quantity FROM ' . _DB_PREFIX_ . 'pack WHERE id_product_pack = ' . (int) $this->id);
        if (!is_array($result))
            $result = [];

        return $result;
    }
```

Reply:
Core team have final word, so let's wait core team review ;)

Goal is only to add phpdoc and not to make code changes because all code change need tests from QA and we don't have enough resources in QA team.

So I'm more in favor of use `mixed` return type here and add a comment with expected result if everything work as expected.

Reply:
Yes, the core team review certainly says the last word, but I want to say something that is reasonably correct. We come together to get feedback and create a quality product.

Reply:
Legacy classes must die but not yet because migration is not achieved.
So we have to do with, invest time to refactor this is waste time and we can't make breaking changes.
Best practice is to use it inside an Adapter to rely on more trusted results and it helps with retrocompatiblity when breaking change has been made in the past but it need an extra work from developers.